lib/krypto/challenge_response.rb

```diff
@@ -28,14 +28,40 @@ def open(private_encryption_key)
         inner = InnerResponse.new(MessagePack.unpack(opened))
 
         # Now that we've opened the box, we can verify that the internal key matches the external signature.
-        public_signing_key = OpenSSL::PKey::EC.new(inner.publicSigningKey)
-        raise "invalid signature" unless Krypto::Ec.verify(public_signing_key, sig, opened)
+        OuterResponse.verify_with_key_bytes(inner.publicSigningKey, sig, opened)
+
+        # If we don't have a signature 2 or a public signing key 2, return what we have
+        if sig2.nil? || sig2.empty?
+          # if there is no sig2, set public signing key 2 to nil just in case so that
+          # the consumer does not fasly assume it was used to perform a signature
+          inner.publicSigningKey2 = nil
+          return inner
+        end
+
+        # have signature but no key
+        if inner.publicSigningKey2.nil? || inner.publicSigningKey2.empty?
+          raise "have signature 2, but no public signing key 2"
+        end
+
+        OuterResponse.verify_with_key_bytes(inner.publicSigningKey2, sig2, opened)
 
         inner
       end
+
+      def self.verify_with_key_bytes(key_bytes, signature, data)
+        if key_bytes.start_with?("-----BEGIN PUBLIC KEY-----")
+          key = OpenSSL::PKey::EC.new(key_bytes)
+          raise "invalid signature" unless Krypto::Ec.verify(key, signature, data)
+          return
+        end
+
+        key = OpenSSL::PKey::EC.new(Base64.strict_decode64(key_bytes))
+        raise "invalid signature" unless Krypto::Ec.verify(key, signature, data)
+        nil
+      end
```

Comment:
If I read this correctly, the only difference is that we don't call `Base64.strict_decode64` for the pem format? (That's a weird library. But 🤷 )

If that's the case, I think this is a lot simpler as:
```
key_bytes = Base64.strict_decode64(key_bytes) unless key_bytes.start_with?("-----BEGIN PUBLIC KEY-----")
key = OpenSSL::PKey::EC.new(key_bytes)
```